### ISSUE TITLE
ImportError: No module named maxflow

### ISSUE THREAD

**Issue by Halimeh-manaf** [loli/medpy#81]
Hi,

so I'm trying to use graphcut of medpy but unfortanually, I'm geting this error. I googled it and there was one issue topic in Github but it suggests that it is a python3 incompatiblity. The thing is, I have python 2.7 installed on my Mac. So I'm not sure what to do now.

This is the error I got: 
 from medpy.graphcut import graphcut_from_voxels
  File "/usr/local/lib/python2.7/site-packages/medpy/graphcut/__init__.py", line 200, in <module>
    from .maxflow import GraphDouble, GraphFloat, GraphInt # this always triggers an error in Eclipse, but is right
ImportError: No module named maxflow

**Comment by loli** [comment]
MedPy can be installed without the graphcut extension. It then signals that the installation of the extension failed during installation.

Could you install medpy again (e.g. in a virtual environment) and post the installation log output here?

**Comment by John1231983** [comment]
I have the same issue. How to install with the virtual environment?

**Comment by loli** [comment]
https://realpython.com/python-virtual-environments-a-primer/

**Comment by John1231983** [comment]
Thanks. This is log
```
john@john:~/python-virtual-environments$ source env/bin/activate
(env) john@john:~/python-virtual-environments$ sudo apt-get install libboost-python-dev build-essential
Reading package lists... Done
Building dependency tree       
Reading state information... Done
build-essential is already the newest version (12.1ubuntu2).
libboost-python-dev is already the newest version (1.58.0.1ubuntu1).
0 upgraded, 0 newly installed, 0 to remove and 28 not upgraded.
(env) john@john:~/python-virtual-environments$ sudo apt-get install python-pip python-numpy python-scipy libboost-python-dev build-essential
Reading package lists... Done
Building dependency tree       
Reading state information... Done
build-essential is already the newest version (12.1ubuntu2).
python-numpy is already the newest version (1:1.11.0-1ubuntu1).
libboost-python-dev is already the newest version (1.58.0.1ubuntu1).
python-scipy is already the newest version (0.17.0-1).
python-pip is already the newest version (8.1.1-2ubuntu0.4).
0 upgraded, 0 newly installed, 0 to remove and 28 not upgraded.
(env) john@john:~/python-virtual-environments$ sudo pip install nibabel pydicom medpy
The directory '/home/john/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
The directory '/home/john/.cache/pip' or its parent directory is not owned by the current user and caching wheels has been disabled. check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
Requirement already satisfied (use --upgrade to upgrade): nibabel in /usr/local/lib/python2.7/dist-packages
Requirement already satisfied (use --upgrade to upgrade): pydicom in /usr/local/lib/python2.7/dist-packages
Requirement already satisfied (use --upgrade to upgrade): medpy in /usr/local/lib/python2.7/dist-packages
Requirement already satisfied (use --upgrade to upgrade): scipy>=0.9.0 in /usr/lib/python2.7/dist-packages (from medpy)
Requirement already satisfied (use --upgrade to upgrade): numpy>=1.6.1 in /usr/local/lib/python2.7/dist-packages (from medpy)
You are using pip version 8.1.1, however version 18.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
(env) john@john:~/python-virtual-environments$ 

```

**Comment by loli** [comment]
You will habe to run pip without sudo. And i just need the log from the
last Fall 'pip installiert medpy'.

**Comment by John1231983** [comment]
I see. That is log
```
pip install nibabel pydicom medpy
Collecting nibabel
  Using cached https://files.pythonhosted.org/packages/a4/63/aad4c6f6ffe3e23226f0905db5994e928e966267d2d191c39e96ddc40bcb/nibabel-2.3.1.tar.gz
Collecting pydicom
  Downloading https://files.pythonhosted.org/packages/f0/8f/0e38dc0bcaf81170e3e6a53c23a0ceb70d9991c7ec887245981b36f18358/pydicom-1.2.1-py2.py3-none-any.whl (7.0MB)
    100% |████████████████████████████████| 7.0MB 10.1MB/s 
Collecting medpy
Collecting numpy>=1.7.1 (from nibabel)
  Downloading https://files.pythonhosted.org/packages/ff/7f/9d804d2348471c67a7d8b5f84f9bc59fd1cefa148986f2b74552f8573555/numpy-1.15.4-cp36-cp36m-manylinux1_x86_64.whl (13.9MB)
    100% |████████████████████████████████| 13.9MB 6.4MB/s 
Collecting six>=1.3 (from nibabel)
  Downloading https://files.pythonhosted.org/packages/73/fb/00a976f728d0d1fecfe898238ce23f502a721c0ac0ecfedb80e0d88c64e9/six-1.12.0-py2.py3-none-any.whl
Collecting scipy>=0.9.0 (from medpy)
  Downloading https://files.pythonhosted.org/packages/a8/0b/f163da98d3a01b3e0ef1cab8dd2123c34aee2bafbb1c5bffa354cc8a1730/scipy-1.1.0-cp36-cp36m-manylinux1_x86_64.whl (31.2MB)
    100% |████████████████████████████████| 31.2MB 2.8MB/s 
Installing collected packages: numpy, six, nibabel, pydicom, scipy, medpy
  Running setup.py install for nibabel ... done
Successfully installed medpy-0.3.0 nibabel-2.3.1 numpy-1.15.4 pydicom-1.2.1 scipy-1.1.0 six-1.12.0
You are using pip version 10.0.1, however version 18.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.

```

**Comment by John1231983** [comment]
This is log
```
g++ is already the newest version (4:5.3.1-1ubuntu1).
libboost-python-dev is already the newest version (1.58.0.1ubuntu1).
Using /home/john/python-virtual-environments/env/lib/python3.6/site-packages
Finished processing dependencies for MedPy==0.3.0
***************************************************************************
WARNING: The medpy.graphcut.maxflow external C++ package could not be compiled, all graphcut functionality will be disabled. You might be missing Boost.Python or some build essentials like g++.
Plain-Python installation succeeded.
***************************************************************************
```
I installed both g++ and libboss

**Comment by loli** [comment]
There seems to be a problem with the dependencies. I'll have to look into
this. Can't tell right now what's the error, sorry.

**Comment by John1231983** [comment]
@loli  I found the error. It cannot find lib boost python 3.6. I search and my PC only has 3.5. How should I fix it?
```
ldconfig -p | grep libboost_python
	libboost_python-py35.so.1.58.0 (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libboost_python-py35.so.1.58.0
	libboost_python-py35.so (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libboost_python-py35.so
	libboost_python-py27.so.1.58.0 (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libboost_python-py27.so.1.58.0
	libboost_python-py27.so (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libboost_python-py27.so

```

**Comment by loli** [comment]
We are getting nearer. From your libboost version (1.58.0.1ubuntu1), I gather you are running an Ubuntu 16.04 LTS. Which comes natively with python3.5 and has a corresponding libboost_python lib.

But for the installation of MedPy, you are using python3.6. Hence the linking version mismatch.

A similar issue is described in : https://github.com/osmcode/pyosmium/issues/52

Good news: I added a work-around for your problem. The setup.py now falls back to the major version libboost link if the major-minor is not found. Fetch the newest master and try again.

This might work, as 3.5 and 3.6 are pretty similar and it's unlikely that the python API has changed in between... but it is possible. I tried it in the other direction and it worked. Fetch the master and try it out for your problem.

Fixed with 4aba755
